### PR TITLE
GH-94597: add deprecation warnings for subclassing `AbstractChildWatcher`

### PR DESCRIPTION
**Pull request by kumaraditya303**
* Issue: gh-94597

### CODE REVIEW

**Comment by kumaraditya303** [comment]
This isn't really needed as deprecation warning will be emitted anyway when configuring the child watcher via `set_child_watcher` but still adding for anyone who is subclassing this but not configures the child watcher for whatever reason.

File: Doc/library/asyncio-policy.rst [hunk 1]
```diff
@@ -222,6 +222,9 @@ implementation used by the asyncio event loop:
       This method has to be called to ensure that underlying
       resources are cleaned-up.
 
+   .. deprecated:: 3.12
```
Comment (gvanrossum) [inline Doc/library/asyncio-policy.rst:225]: (How sure are you that this placement will make it apply to the whole class instead of just to `close()`?)

**Review by gvanrossum — APPROVED** [review]
(Do double check the appearance of the deprecation note.)

File: Doc/library/asyncio-policy.rst (hunk 1 above)
Comment (kumaraditya303) [reply Doc/library/asyncio-policy.rst:225]: image

It applies to the whole class.

**Pull request merged**